docs/tinlake/contracts/contracts.md

```diff
@@ -72,25 +72,30 @@ The method `accrue(uint loan)` needs to be called by the Shelf before any modifi
+1) Principal: The Principal ceiling let's a loan borrow up to a given amount exactly once. Wards can set different amounts for each loan using the `file` method. When the user wants to borrow, the amount will be deducted from the user's principal.  
+2) Creditline: This ceiling contract keeps track of a credit limit. It will allow any borrow request to go through provided sum of the requested amount and the debt reported by the Pile for the loan is not greater than the credit limit. The credit limit can be set by a ward on the contract using the `file` method.  
+
+### Threshold
+
+The `Threshold` contract is essential to keep track of the loans' health. For each loan it stores the maximum amount the loan debt is allowed to reach before a loan is considered undercollaterized.
+The threshold can be set by a ward on the contract using the `file` method.
 
 ### Collector
-The Collector contract handles the collection of undercollaterized loans. If the Debt of a loan is larger than the Treshold, Collector then allows Liquidators/Keepers to collect the underlying NFTs from Tinlake.
+The `Collector` contract handles the collection of undercollaterized loans. If the Debt of a loan is larger than the Threshold, Collector then allows Liquidators/Keepers to collect the underlying NFTs from Tinlake.
 
-To initiate the collection, any user can call `seize` on the Collector. The Collector then calls `get` on Threshold. If Debt < Threshold, Collector aborts the action. If Debt > Threshold, Collector calls `claim` to move the NFT from the Shelf to the Collector. From there, Liquidators can collect the NFT at a price set by a service provider. Note, that only whitelisted Liquidators can call `collect`.
+To initiate the collection, any user can call `seize` on the Collector. The Collector then calls `get` on Threshold. If ```Debt < Threshold```, Collector aborts the action. If ```Debt > Threshold```, Collector calls `claim` to move the NFT from the Shelf to the Collector. From there, Liquidators can collect the NFT at a price set by a service provider. Note, that only whitelisted Liquidators can call `collect`.
 
 ## Lender Contracts
-The Lender Contracts interact with the borrower side by supplying an amount of Currency ERC20 as requested by the Shelf.balanceRequest() method.
+The Lender Contracts interact with the borrower side by supplying an amount of Currency ERC20 as requested by the `Shelf.balanceRequest()` method.
 
 ![Lender Contracts Call Graph](../../../src/images/tinlake/lender.svg)
 
 
-Tinlake enables lenders to invest two differenct tranches: A senior tranche issuing a token called DROP, and a junior tranche issuing a token called TIN. This structure is similar to [senior/junior structures in finance](blah).
+Tinlake enables lenders to invest in two differenct tranches: A senior tranche issuing a token called DROP, and a junior tranche issuing a token called TIN. This structure is similar to [senior/junior structures in finance](blah).
```

Comment:
I assume, (blah_ shoud be alink to tindrop.md?)

Reply:
This is a good question :D

Reply:
Pretty sure is should ;-). Let's either delete or add the link